test/images/agnhost/api-coherent/main.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apicoherent
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+)
+
+var (
+	apiName  string
+	contains []string
+	excludes []string
+	status   int32
+	timeout  int
+)
+
+type response struct {
+	apiServer string
+	response  string
+	err       error
+}
+
+// CmdAPICoherent is used by agnhost Cobra.
+var CmdAPICoherent = &cobra.Command{
+	Use:   "api-coherent",
+	Short: "Checks for a coherent response against all apiserver instances when asking for specific API groups",
+	Long:  `Checks for a coherent response against all apiserver instances using the in-cluster config. Because of this, this subcommand is meant to be run inside of a Kubernetes pod.`,
+	Args:  cobra.MaximumNArgs(0),
+	Run:   main,
+}
+
+func init() {
+	CmdAPICoherent.Flags().StringVar(&apiName, "api", "", "API group to ask for. E.g. rbac.authorization.k8s.io/v1")
+	CmdAPICoherent.Flags().StringSliceVar(&contains, "contains", nil, "Strings that must be included in the response. List.")
+	CmdAPICoherent.Flags().StringSliceVar(&excludes, "excludes", nil, "Strings that mustnt be included in the response. List.")
+	CmdAPICoherent.Flags().IntVar(&timeout, "timeout", 60, "Time before bailing out. Seconds.")
+	CmdAPICoherent.Flags().Int32Var(&status, "status", 200, "HTTP status to expect from the response. E.g. 200(success), 404(not found)")
+}
+
+func main(cmd *cobra.Command, args []string) {
+	logs.InitLogs()
+	defer logs.FlushLogs()
+
+	flag.Set("logtostderr", "true")
+
+	klog.Infof("Started")
+	klog.Infof("API to query: %s", apiName)
+	klog.Infof("Response must contain: %v", contains)
+	klog.Infof("Response must not contain: %v", excludes)
+	klog.Infof("Timeout before failing: %d", timeout)
+	klog.Infof("Expected HTTP status code: %d", status)
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Unable to get in-cluster config: %v", err)
+	}
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		log.Fatalf("Unable to create client; error: %v\n", err)
```

Comment:
Could you change this to use klog instead?

Reply:
Done